NLog.Windows.Forms/RichTextBoxTarget.cs

```diff
@@ -299,6 +318,100 @@ public RichTextBoxTargetMessageRetentionStrategy MessageRetention
+        /// <seealso cref="LinkClicked"/>
+        /// </summary>
+        [DefaultValue(false)]
+        public bool SupportLinks 
+        {
+            get { return supportLinks; }
+            set
+            {
+                supportLinks = value;
+                if (supportLinks)
+                {
+                    lock (linkedEventsLock)
+                    {
+                        if (linkedEvents == null)
+                        {
+                            linkedEvents = new Dictionary<int, LogEventInfo>();
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Type of delegate for <see cref="LinkClicked"/> event.
+        /// </summary>
+        /// <param name="sender">The target that caused the event</param>
+        /// <param name="linkText">Visible text of the link being clicked</param>
+        /// <param name="logEvent">Original log event that caused a line with the link</param>
+        public delegate void DelLinkClicked(RichTextBoxTarget sender, string linkText, LogEventInfo logEvent);
+
+        /// <summary>
+        /// Event fired when the user clicks on a link in the control created by the "rtb-link" renderer (<see cref="RichTextBoxLinkLayoutRenderer"/>).
+        /// <seealso cref="DelLinkClicked"/>
+        /// </summary>
+        public event DelLinkClicked LinkClicked;
+
+        /// <summary>
+        /// Actual value of the <see cref="LinkClicked"/> property
+        /// </summary>
+        private bool supportLinks = false;
+
+        /// <summary>
+        /// Lock for <see cref="linkedEvents"/> dictionary access
+        /// </summary>
+        private readonly object linkedEventsLock = new object();
+
+        /// <summary>
+        /// A map from link id to a corresponding log event
+        /// </summary>
+        private Dictionary<int, LogEventInfo> linkedEvents;
+
+        /// <summary>
+        /// Returns number of events stored for active links in the control. 
+        /// Used only in tests to check that non needed events are removed.
+        /// </summary>
+        internal int? LinkedEventsCount
+        {
+            get
+            {
+                lock (linkedEventsLock)
+                {
+                    if (linkedEvents == null)
+                    {
+                        return null;
+                    }
+                    return linkedEvents.Count;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Internal prefix that is added to the link id in RTF
+        /// </summary>
+        private const string LINK_PREFIX = "link";
+
+
+        /// <summary>
+        /// Used to capture link placeholders in <see cref="SendTheMessageToRichTextBox"/>
+        /// </summary>
+        private readonly Regex linkAddRegex = new Regex(@"(\([a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}\))", RegexOptions.Compiled);
```

Comment:
I have seen a lot of memory issues with compiled regexes. I have read a lot of it in the past, and I did re-read your links. Also find this useful: http://stackoverflow.com/questions/9969158/when-not-to-use-regexoptions-compiled

I think the we can use compile, iif:

- The regex is compiled, eg. use `static` (and preferred, lazy)
- The regex is used a lot. This is the case for 'linkAddRegex', and 'linkRemoveRtfRegex', but not for `linkClickRegex`
- If performance is important, don't use regexes if that's is possible
- and of course prevent backtracking. 

So we can keep two of them, if there will be static.